heat/core/logical.py

```diff
@@ -307,3 +314,55 @@ def logical_xor(t1, t2):
     tensor([ False, False,  True])
     """
     return operations.__binary_op(torch.logical_xor, t1, t2)
+
+
+def _sanitize_close_input(x, y):
+    if np.isscalar(x):
+        try:
+            if isinstance(y, dndarray.DNDarray):
+                x = factories.array(float(x), device=y.device)
+            else:
+                x = factories.array(float(x))
+        except (ValueError, TypeError):
+            raise TypeError("Data type not supported, input was {}".format(type(x)))
+
+    elif not isinstance(x, dndarray.DNDarray):
+        raise TypeError(
+            "Only tensors and numeric scalars are supported, but input was {}".format(type(x))
+        )
+
+    if np.isscalar(y):
+        try:
+            if isinstance(x, dndarray.DNDarray):
+                y = factories.array(float(y), device=x.device)
+            else:
+                y = factories.array(float(y))
+        except (ValueError, TypeError):
+            raise TypeError("Data type not supported, input was {}".format(type(y)))
+
+    elif not isinstance(y, dndarray.DNDarray):
+        raise TypeError(
+            "Only tensors and numeric scalars are supported, but input was {}".format(type(y))
+        )
+
+    # Do redistribution out-of-place
+    # If only one of the tensors is distributed, unsplit/gather it
+    if (x.split is not None) and (y.split is None):
+        t1 = manipulations.resplit(x, axis=None)
+        t2 = y.copy()
+
+    elif (x.split is None) and (y.split is not None):
+        t1 = x.copy()
+        t2 = manipulations.resplit(y, axis=None)
+
+    # If both x and y are split, but along different axes, y is redistributed to be split along the same axis as x
+    # TODO: needs broadcasting first depending on shapes
+    elif (x.split is not None) and (y.split is not None) and (x.split != y.split):
+        t1 = x.copy()
+        t2 = manipulations.resplit(y, axis=x.split)
+
+    else:
+        t1 = x.copy()
+        t2 = y.copy()
+
```

Comment:
there are lots of copy calls here. do we need to copy the data? do any of the 'close' functions overwrite the data?

Reply:
They don't overwrite the data, but they do resplit the tensors in some cases. 

The last conditional statement doesn't need to copy though, I don't think. I've changed it to simply return x, y.